src/generators/notebookgenerator/index.ts

```diff
@@ -566,23 +238,141 @@ namespace Private {
     return 0;
   }
 
+  export function processMD(
```

Comment:
This takes and returns a kind of long set of stuff. Can you document the intention of all the arguments and return values?